A27-xds-global-load-balancing.md

```diff
@@ -0,0 +1,430 @@
+the xDS server will be contained within an XdsClient object, which will
+be used by both the resolver and the LB policies.
+
+There will be two separate LB policies for xDS, one to support `Cluster`
+data, and another to support `ClusterLoadAssignment` data.
+
+Note that the LB policies will have "experimental" suffixes for their
+names for now.  These suffixes will be removed when the functionality
+has proven to be stable.
+
+![gRPC Client Architecture Diagram](A27_graphics/grpc_client_architecture.png)
+
+[Link to SVG file](A27_graphics/grpc_client_architecture.svg)
+
+#### XdsClient and Bootstrap File
+
+The XdsClient object contains all of the logic for interacting with the
+xDS server.  The XdsClient object is a purely internal API, not
+something exposed to gRPC users, but it is described here as part of
+the architecture used to support xDS in the gRPC client.
+
+The XdsClient object is configured via a bootstrap file.  The location
+of the bootstrap file is determined via the `GRPC_XDS_BOOTSTRAP`
+environment variable.  The file is in JSON form, and its contents look
+like this:
+
+```
+{
+  // The xDS server to talk to.  The value is an array to allow for a
+  // future change to add support for failing over to a secondary xDS server
+  // if the primary is down, but for now, only the first entry in the
+  // array will be used.
+  xds_servers": [
+    {
+      "server_uri": <string containing URI of xds server>,
+      // List of channel creds; client will stop at the first type it
+      // supports.  This field is optional; if not specified, xDS will use
+      // the same channel creds as the backends, but with any associated
+      // call creds stripped off.
+      "channel_creds": [
+        {
+          "type": <string containing channel cred type>,
+          // The "config" field is optional; it may be missing if the
+          // credential type does not require config parameters.
+          "config": <JSON object containing config for the type>
+        }
+      ]
+    }
+  ],
+  "node": <JSON form of Node proto>
+}
+```
+
+Initially, the only type of channel creds we support will be
+`google_default`.  In the future, we will add a general-purpose
+mechanism for configuring arbitrary channel creds types with arbitrary
+configuration.
+
+The `node` field will be populated with the [xDS `Node`
+proto](https://github.com/envoyproxy/data-plane-api/blob/1adb5d54abb0e28ca409254d26fad1cf5535239b/envoy/api/v2/core/base.proto#L85).
+Note that the `build_version` field should not be specified in the
+bootstrap file, since that will be populated automatically based on the gRPC
+client's version.
+
+To allow for future changes in a backward-compatible way, unknown fields
+in the bootstrap file will be silently ignored.
+
+#### xds Resolver
+
+Clients will enable use of xDS by using the "xds" resolver in the
+target URI used to create the gRPC channel.  For example, a user may
```

Comment:
That probably would have worked for just the specific set of functionality that we're going to provide in the initial release, but we are definitely going to need the full xDS API flow later when we add support for things like routing and configurable LB policies, and we wanted the right structure in place from the get-go so that users don't have to change things on their clients or their management servers later as we add more features.